modules/core/src/main/java/org/apache/ignite/spi/communication/tcp/TcpCommunicationSpi.java

```diff
@@ -3178,6 +3214,30 @@ private void checkClientQueueSize(GridNioSession ses, int msgQueueSize) {
                 log.debug("Addresses to connect for node [rmtNode=" + node.id() + ", addrs=" + addrs + ']');
         }
 
+        // Apply the configured node address exclusion filters, if any.
+        if(!this.getNodeAddressExclusionFilters().isEmpty()) {
+           final LinkedHashSet<InetSocketAddress> filteredAddrs = new LinkedHashSet<>();
+           for (final InetSocketAddress addr : addrs) {
+              for (final String filter : this.getNodeAddressExclusionFilters()) {
+                 if (addr.getAddress().getHostAddress().matches(filter)) {
```

Comment:
What are we intend to do with invalid regular expression PatternSyntaxException?

Reply:
Yes, I would add a check for the valid regular expression to setter and log invalid settings. If we do so we can also save Pattern.compile as a member variable and use for matching.